custom_components/tuya_local/devices/agl_ultracontato.yaml

```diff
@@ -0,0 +1,84 @@
+name: AGL ultracontato r2
+primary_entity:
+  entity: cover
+  class: door
+  dps:
+    - id: 1
+      name: control
+      type: boolean
+      mapping:
+        - dps_val: true
+          value: "open"
+        - dps_val: false
+          value: "close"
+    - id: 101
+      name: open
+      type: boolean
+secondary_entities:
+  - entity: select
+    name: Switch type
+    category: config
+    icon: "mdi:electric-switch-closed"
+    dps:
+      - id: 108
+        type: string
+        name: option
+        mapping:
+          - dps_val: "NA"
+            value: Normally Open
+          - dps_val: "NF"
+            value: Normally Closed
+  - entity: select
+    category: config
+    name: Pulse type
+    icon: "mdi:pulse"
+    dps:
+      - id: 109
+        name: option
+        type: string
+        mapping:
+          - dps_val: "Pulsado"
+            value: Pulsed
+          - dps_val: "Retencao"
+            value: Retencion
+          - dps_val: "ONOFF"
+            value: ON - OFF
+  - entity: select
+    category: config
+    name: Autoclose
+    icon: "mdi:door-closed"
+    dps:
+      - id: 103
+        name: option
+        type: string
+        mapping:
+          - dps_val: "habilitado"
+            value: Enabled
+          - dps_val: "desabilitado"
+            value: Disabled
+  - entity: number
+    category: config
+    name: During time
+    icon: "mdi:lock-clock"
+    dps:
+      - id: 107
+        name: value
+        type: integer
+        range:
+          min: 1
+          max: 65
+        unit: s
+  - entity: binary_sensor
+    name: Door Opened
+    category: diagnostic
+    dps:
+      - id: 111
+        type: boolean
+        name: sensor
+  - entity: binary_sensor
```

Comment:
It seems strange to have door opened and door closed as different sensors.  Is one of them the lock state?  In which case the class should be "lock" and no name is needed.

But there is also an additional open sensor on dp 101 used in the cover entity.